src/Type/PHPUnit/CreateMockDynamicReturnTypeExtension.php

```diff
@@ -38,25 +39,13 @@ public function getTypeFromMethodCall(MethodReflection $methodReflection, Method
 		if (!isset($methodCall->args[$argumentIndex])) {
 			return $methodReflection->getReturnType();
 		}
-		$arg = $methodCall->args[$argumentIndex]->value;
-		if (!($arg instanceof \PhpParser\Node\Expr\ClassConstFetch)) {
-			return $methodReflection->getReturnType();
-		}
 
-		$class = $arg->class;
-		if (!($class instanceof \PhpParser\Node\Name)) {
+		$argType = $scope->getType($methodCall->args[$argumentIndex]->value);
+		if (!($argType instanceof TypeWithClassName)) {
```

Comment:
This doesn't look right to me - `Foo::class` does not result in `TypeWithClassName` but in `ConstantStringType`. Also, this does not break on dev-master so I don't see a reason to change it.

Reply:
And you are right of course. Will fix that.